src/analysis/calculus/times_cont_diff.lean

```diff
@@ -1519,24 +1790,16 @@ begin
   /- we lift all the spaces to a common universe, as we have already proved the result in this
   situation. For the lift, we use the trick that `H` is isomorphic through a
   continuous linear equiv to `continuous_multilinear_map 𝕜 (λ (i : fin 0), (E × F × G)) H`, and
-  continuous linear equivs respect smoothness classes.
-  The instances are not found automatically by Lean, so we declare them by hand.
-  TODO: fix. -/
+  continuous linear equivs respect smoothness classes. -/
   let Eu := continuous_multilinear_map 𝕜 (λ (i : fin 0), (E × F × G)) E,
-  letI : normed_group Eu :=
-    @continuous_multilinear_map.to_normed_group 𝕜 (fin 0) (λ (i : fin 0), E × F × G) E _ _ _ _ _ _ _,
-  letI : normed_space 𝕜 Eu :=
-    @continuous_multilinear_map.to_normed_space 𝕜 (fin 0) (λ (i : fin 0), E × F × G) E _ _ _ _ _ _ _,
+  letI : normed_group Eu := by apply_instance,
+  letI : normed_space 𝕜 Eu := by apply_instance,
```

Comment:
What's happening here? Is this done to somehow raise the priority of instances in type class searches down the line?

Reply:
Before, Lean could not find the instance by itself, so one had to give it explicitly. With the improvements done by Gabriel over the last few months, Lean has become better at this, so `apply_instance` is enough. But Lean is not yet awesome enough to remove the instances: if you don't give them, then a rewrite below fails. I guess this is due to the fact that unification with Pi types is complicated, so Lean needs some help.